deployment/ccip/changeset/v1_5_1/cs_add_token_e2e.go

```diff
@@ -39,96 +40,106 @@ import (
+	// (ex: override to clCCIP-LnM when the main token symbol is CCIP-LnM)
+	// WARNING: This should only be used in exceptional cases where the token symbol on
+	// a particular chain differs from the main tokenSymbol
+	OverrideTokenSymbol shared.TokenSymbol `json:"overrideTokenSymbol,omitempty"`
 }
 
 type AddTokenE2EConfig struct {
-	PoolConfig   map[uint64]E2ETokenAndPoolConfig
-	IsTestRouter bool
+	// Map of chain ID to E2ETokenAndPoolConfig.
+	PoolConfig map[uint64]E2ETokenAndPoolConfig `json:"poolConfig"`
+
+	// Whether this is a test router configuration.
+	IsTestRouter bool `json:"isTestRouter"`
+
+	// Configures the pools, if empty deployed pools aren't configured.
+	ConfigurePools ConfigureTokenPoolContractsConfig `json:"configurePools"`
 
 	// internal fields - To be populated from the PoolConfig.
-	// User do not need to populate these fields.
+	// User does not need to populate these fields.
 	deployPool             *DeployTokenPoolContractsConfig
-	configurePools         ConfigureTokenPoolContractsConfig
 	configureTokenAdminReg TokenAdminRegistryChangesetConfig
 }
 
 // newConfigurePoolAndTokenAdminRegConfig populated internal fields in AddTokenE2EConfig.
 // It creates the configuration for deploying and configuring token pools and token admin registry.
 // It then validates the configuration.
 func (c *AddTokenE2EConfig) newConfigurePoolAndTokenAdminRegConfig(e cldf.Environment, symbol shared.TokenSymbol, timelockCfg *proposalutils.TimelockConfig, existingPool bool) error {
-	if !existingPool {
-		c.deployPool = &DeployTokenPoolContractsConfig{
-			TokenSymbol:  symbol,
-			NewPools:     make(map[uint64]DeployTokenPoolInput),
-			IsTestRouter: c.IsTestRouter,
-		}
-	}
-	c.configurePools = ConfigureTokenPoolContractsConfig{
-		TokenSymbol: symbol,
-		MCMS:        nil, // as token pools are deployed as part of the changeset, the pools will still be owned by the deployer key
-		PoolUpdates: make(map[uint64]TokenPoolConfig),
+
+	c.deployPool = &DeployTokenPoolContractsConfig{
+		TokenSymbol:  symbol,
+		NewPools:     make(map[uint64]DeployTokenPoolInput),
+		IsTestRouter: c.IsTestRouter,
 	}
+
+	// ensuring already existing pool has ownership transferred to MCMs
 	if existingPool {
-		c.configurePools.MCMS = timelockCfg
+		c.ConfigurePools.MCMS = timelockCfg
 	}
 	c.configureTokenAdminReg = TokenAdminRegistryChangesetConfig{
 		MCMS:  timelockCfg,
 		Pools: make(map[uint64]map[shared.TokenSymbol]TokenPoolInfo),
 	}
 	for chain, poolCfg := range c.PoolConfig {
-		tpCfg := TokenPoolConfig{
-			ChainUpdates:        poolCfg.RateLimiterConfig,
-			Version:             poolCfg.PoolVersion,
-			OverrideTokenSymbol: poolCfg.OverrideTokenSymbol,
-			SolChainUpdates:     poolCfg.SolChainUpdates,
-		}
+		tpCfg := TokenPoolConfig{}
 		poolInfo := TokenPoolInfo{
 			Version:       poolCfg.PoolVersion,
 			ExternalAdmin: poolCfg.ExternalAdmin,
 		}
-		if !existingPool {
+
+		if poolCfg.ExistingPoolType == nil {
```

Comment:
@AnieeG had to make this change based on your DMs 

tldr 

We were looping over `c.PoolConfig` and using the `existingPool` flag to decide which configuration path to take:

But existingPool is passed as a single boolean for the entire token, not per-chain, even though `c.PoolConfig` is per-chain, from here (https://github.com/smartcontractkit/chainlink/blob/develop/deployment/ccip/changeset/v1_5_1/cs_add_token_e2e.go#L249-L253)